src/participant_network/cl/lighthouse/lighthouse_launcher.star

```diff
@@ -0,0 +1,300 @@
+
+	if mev_boost_context != None:
+		cmd_args.append("--builder")
+		cmd_args.append(mev_boost_endpoint(mev_boost_context))
+
+
+	if len(extra_params) > 0:
+		# this is a repeated<proto type>, we convert it into Starlark
+		cmd_args.extend([param for param in extra_params])
+
+
+	return struct(
+		image = image,
+		ports = BEACON_USED_PORTS,
+		cmd_args = cmd_args,
+		files = {
+			genesis_data.files_artifact_uuid: GENESIS_DATA_MOUNTPOINT_ON_CLIENTS
+		},
+		env_vars = {
+			RUST_BACKTRACE_ENVVAR_NAME: RUST_FULL_BACKTRACE_KEYWORD
+		},
+		privaite_ip_address_placeholder = PRIVATE_IP_ADDRESS_PLACEHOLDER
+	)
+
+
+def get_validator_config(
+	genesis_data,
+	image,
+	log_level,
+	beacon_client_http_url,
+	node_keystore_files,
+	mev_boost_context,
+	extra_params):
+
+	# For some reason, Lighthouse takes in the parent directory of the config file (rather than the path to the config file itself)
+	genesis_config_parent_dirpath_on_client = path_join(GENESIS_DATA_MOUNTPOINT_ON_CLIENTS, path_dir(genesis_data.config_yml_rel_filepath))
+	validator_keys_dirpath = path_join(VALIDATOR_KEYS_MOUNTPOINT_ON_CLIENTS, node_keystore_files.raw_keys_relative_dirpath)
+	validator_secrets_dirpath = path_join(VALIDATOR_KEYS_MOUNTPOINT_ON_CLIENTS, node_keystore_files.raw_secrets_relative_dirpath)
+	
+	cmd_args = [
+		"lighthouse",
+		"validator_client",
+		"--debug-level=" + log_level,
+		"--testnet-dir=" + genesis_config_parent_dirpath_on_client,
+		"--validators-dir=" + validator_keys_dirpath,
+		# NOTE: When secrets-dir is specified, we can't add the --data-dir flag
+		"--secrets-dir=" + validator_secrets_dirpath,
+		# The node won't have a slashing protection database and will fail to start otherwise
+		"--init-slashing-protection",
+		"--http",
+		"--unencrypted-http-transport",
+		"--http-address=0.0.0.0",
+		"--http-port={0}".format(VALIDATOR_HTTP_PORT_NUM),
+		"--beacon-nodes=" + beacon_client_http_url,
+		#"--enable-doppelganger-protection", // Disabled to not have to wait 2 epochs before validator can start
+		# burn address - If unset, the validator will scream in its logs
+		"--suggested-fee-recipient=0x0000000000000000000000000000000000000000",
+		# vvvvvvvvvvvvvvvvvvv PROMETHEUS CONFIG vvvvvvvvvvvvvvvvvvvvv
+		"--metrics",
+		"--metrics-address=0.0.0.0",
+		"--metrics-allow-origin=*",
+		"--metrics-port={0}".format(VALIDATOR_METRICS_PORT_NUM),
+		# ^^^^^^^^^^^^^^^^^^^ PROMETHEUS CONFIG ^^^^^^^^^^^^^^^^^^^^^
+	]
+
+	if mev_boost_context != None:
+		cmd_args.append("--builder-proposals")
+
+	if len(extra_params):
+		cmd_args.extend([param for param in extra_params])
+
+
+	return struct(
+		image = image,
+		ports = VALIDATOR_USED_PORTS,
+		cmd_args = cmd_args,
+		files = {
+			genesis_data.files_artifact_uuid: GENESIS_DATA_MOUNTPOINT_ON_CLIENTS,
+			node_keystore_files.files_artifact_uuid: VALIDATOR_KEYS_MOUNTPOINT_ON_CLIENTS,
+		},
+		env_vars = {
+			RUST_BACKTRACE_ENVVAR_NAME: RUST_FULL_BACKTRACE_KEYWORD
+		},
```

Comment:
why don't we set the private ip placeholder here?

Reply:
none of the cmd args, env vars or entrypoint args use the place holder, so it needs no replacement. It picks the default which is fine.